cloud/openstack/os_attach_volume

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from ansible.module_utils.basic import *
+from novaclient.client import Client
+from cinderclient import client as cinderclient
+from retrying import retry
+import os
+
+DOCUMENTATION =  '''
+---
+module: os_attach_volume
+version_added: "1.9"
+short_description: Attach a volume to an instance
+description:
+   - Create and attach a volume to an instance
+author: Max Cameron
+options:
+  login_username:
+    description:
+      - login username to authenticate to keystone
+    required: true
+    default: admin
+  login_password:
+    description:
+      - Password of login user
+    required: true
+    default: 'yes'
+  login_tenant_name:
+    description:
+      - The tenant name of the login user
+    required: true
+    default: 'yes'
+  auth_url:
+    description:
+      - The keystone url for authentication
+    required: false
+    default: 'http://127.0.0.1:35357/v2.0/'
+  region_name:
+    description:
+      - Name of the region
+    required: false
+    default: None
+  name:
+    description:
+      - Name that has to be given to the instance (Volume will inherit the name)
+    required: true
+    default: None
+  size:
+    description:
+      - Size of the volume to attach
+    required: True
+    default: None
```

Comment:
if anything is required, there is no default, otherwise it would not be required.